api/v1beta1/configuration_types.go

```diff
@@ -67,6 +67,9 @@ type BaseConfigurationSpec struct {
 
 	// Region is cloud provider's region. It will override the region in the region field of ProviderReference
 	Region string `json:"region,omitempty"`
+
+	// GitCredentialsReference specifies the reference to the secret containing the git credentials
+	GitCredentialsReference *types.SecretReference `json:"gitCredentialsReference,omitempty"`
```

Comment:
can we move this field into the ConfigurationSpec outside from `BaseConfigurationSpec`, then it won't affect the CRD in KubeVela repo.